### PR TITLE
libutils: brel: bugfix memset scope larger than expected

### PR DESCRIPTION
**Pull request by zjuzzy**
In the case of wipe = true and previous buffer is free,    memset
the current buffer, rather then current buffer with previous free
buffer.

Signed-off-by: Zengxx <zengxiaoxu@huawei.com>

For more details, please see:(https://github.com/OP-TEE/optee_os/issues/5542)

### CODE REVIEW

**Review by etienne-lms — COMMENTED** [review]
~~When previous buffer is free, why do you want to **not** wipe its content? Has the content already been wiped from bget previosuly?~~

@zjuzzy, I should have better looked, I see your point. Indeed as freed buffer is at its default state or has already been freed. Discard my question. :confused:

File: lib/libutils/isoc/bget.c [hunk 1]
```diff
@@ -983,6 +985,8 @@ void brel(buf, poolset, wipe)
 	   allocated. */
 
 	register bufsize size = b->bh.bsize;
+        wipe_start = (char *)b;
```
Comment (etienne-lms) [inline lib/libutils/isoc/bget.c:988]: I sugget to add a inline comment, like
`/* Only current buffer content, including bfhead, needs to be wiped */`

File: lib/libutils/isoc/bget.c [hunk 2]
```diff
@@ -1028,10 +1034,10 @@ void brel(buf, poolset, wipe)
 	   memory.  */
 
 	bn = BFH(((char *) b) + b->bh.bsize);
+        wipe_size = wipe_size + sizeof(struct bfhead);
```
Comment (etienne-lms) [inline lib/libutils/isoc/bget.c:1037]: ditto:

```c
	/* Only bfhead of next buffer needs to be wiped */
	wipe_size += sizeof(struct bfhead);

File: lib/libutils/isoc/bget.c [hunk 3]
```diff
@@ -1028,10 +1034,10 @@ void brel(buf, poolset, wipe)
...
     }
     if (wipe) {
-	V memset_unchecked(((char *) b) + sizeof(struct bfhead), 0x55,
-			   (MemSize) (b->bh.bsize - sizeof(struct bfhead)));
+	V memset_unchecked(wipe_start, 0x55, (MemSize)wipe_size);
```
Comment (etienne-lms) [inline lib/libutils/isoc/bget.c:1040]: as `wipe_start` as no default init value, I would add here `assert(wipe_start);`.

File: lib/libutils/isoc/bget.c [hunk 4]
```diff
@@ -1000,6 +1004,8 @@ void brel(buf, poolset, wipe)
 	poolset->freelist.ql.blink = b;
 	b->ql.blink->ql.flink = b;
 	b->bh.bsize = -b->bh.bsize;
+        wipe_start = (char *)b + sizeof(struct bfhead);
```
Comment (etienne-lms) [inline lib/libutils/isoc/bget.c:1007]: Prefer add an empty line above.
Maybe add a dedicated inline comment for the added instructions? (`/* Set memory area to be wiped */`)?

**Review by etienne-lms — COMMENTED** [review]
indeed, your change looks consistent to me.

File: lib/libutils/isoc/bget.c (hunk 3 above)
Comment (jenswi-linaro) [reply lib/libutils/isoc/bget.c:1040]: > as `wipe_start` as no default init value, I would add here `assert(wipe_start);`.

That seems pointless. What would you be asserting?

Comment (jenswi-linaro) [inline lib/libutils/isoc/bget.c:1040]: The cast isn't needed `wipe_size` already has this type.

File: lib/libutils/isoc/bget.c (hunk 2 above)
Comment (zjuzzy) [reply lib/libutils/isoc/bget.c:1037]: OKey.

**Review by jenswi-linaro — COMMENTED** [review]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by jforissier** [comment]
Please make the subject: `libutils: brel: bugfix memset scope larger than expected`.

File: lib/libutils/isoc/bget.c (hunk 1 above)
Comment (zjuzzy) [reply lib/libutils/isoc/bget.c]: Okey, Comments have been added.

File: lib/libutils/isoc/bget.c (hunk 4 above)
Comment (zjuzzy) [reply lib/libutils/isoc/bget.c]: Fine, I've added an empty line above.

File: lib/libutils/isoc/bget.c (hunk 3 above)
Comment (zjuzzy) [reply lib/libutils/isoc/bget.c]: I've removed this cast.

**Comment by zjuzzy** [comment]
> > @[zjuzzy](https://github.com/zjuzzy) zjuzzy changed the title lib:brel:bugfix memset scope larger than expected. libutils: brel: bugfix memset scope larger than expected
> 
> The commit subject, not the PR subject (sorry).

I've changed the PR subject. (THANKS!)

**Comment by zjuzzy** [comment]
> `Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`Thank you for your review(respect!).

File: lib/libutils/isoc/bget.c [hunk 5]
```diff
@@ -984,6 +986,10 @@ void brel(buf, poolset, wipe)
 
 	register bufsize size = b->bh.bsize;
 
+	/* Only wipe the current buffer, including bfhead.*/
```
Comment (etienne-lms) [inline lib/libutils/isoc/bget.c:989]: add a space char before `*/`

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` with a minor comment.

**Comment by etienne-lms** [comment]
Byt the way, you can drop the period at end of header line:
```patch
-libutils: brel: bugfix memset scope larger than expected.
+libutils: brel: bugfix memset scope larger than expected

**Review by wgzh1987 — APPROVED** [review]
Reviewed-by: Guangwei Zhou <zhouguangwei5@huawei.com>

File: lib/libutils/isoc/bget.c (hunk 5 above)
Comment (zjuzzy) [reply lib/libutils/isoc/bget.c]: done(THANKS!)

**Pull request merged**